src/controllers/user-posts.js

```diff
@@ -0,0 +1,179 @@
+const prisma = require("../utils/prisma.js");
+const jwt = require('jsonwebtoken');
+
+
+const { MissingFieldsError, CantFindIdError, InvalidTokenError, UnauthorizedError } = require("../utils/errors");
+const { post, user } = require("../utils/prisma.js");
+
+const createPost = async (req, res) => {
+  const token = req.get('authorization');
+  const userId = Number(req.params.userId);
+  let payload = ''
+  console.log('token: ',token)
+
+  try {
+    payload = Number(jwt.verify(token, process.env.JWT_SECRET_KEY))
```

Comment:
This will need to change once you address the payload comment above

Reply:
```js
const token = req.get('authorization');
  const userId = Number(req.params.userId);
  let payload = ''

  try {
    payload = jwt.verify(token, process.env.JWT_SECRET_KEY)
  } catch (error) {
    throw new InvalidTokenError()
  }
  console.log(payload.id, userId)
  if (payload.id !== userId) {
    throw new UnauthorizedError()
  }
  ```
  Fixed!